File: Javascript/tire-pressure-monitoring-system/alarm.js
var Sensor = require('./sensor.js');

Alarm = function() {
	this._lowPressureThreshold = 17;
	this._highPressureThreshold = 21;
	this._sensor = new Sensor();
	this._alarmOn = false;
};

Alarm.prototype = {

	check: function() {
		var psiPressureValue = this._sensor.popNextPressurePsiValue();

		if (psiPressureValue < this._lowPressureThreshold || this._highPressureThreshold < psiPressureValue) {
			this._alarmOn = true;
		}
	},

	alarmOn: function() {
		return this._alarmOn;
	}

};

module.exports = Alarm;

File: go/tire-pressure-monitoring-system/alarm.go
package tpms

type Alarm interface {
	check()
}

type Sensor interface {
	popNextPressurePsiValue() int
}

type alarm struct {
	lowPressureThreshold  int
	highPressureThreshold int
	sensor                Sensor
	alarmOn               bool
}

func (a *alarm) check() {
	p := a.sensor.popNextPressurePsiValue()

	if p < a.lowPressureThreshold || a.highPressureThreshold < p {
		a.alarmOn = true
	}
}

func NewAlarm() Alarm {
	return &alarm{
		lowPressureThreshold:  17,
		highPressureThreshold: 21,
		alarmOn:               false,
		sensor:                NewSensor(),
	}

}
